File: iosApp/iosApp/ui/userData/UserDataViewModel.swift
import Foundation
import shared

extension UserDataScreen {
    class UserDataViewModel : ObservableObject, ViewLifeCycleObserver {
        
        private var commonViewModel: UserDataCommonViewModel?
        
        @Published var userLoadStatus: LoadStatus<User>? = nil
        
        func startObserving() {
            let commonViewModel = UserDataCommonViewModel()
            self.commonViewModel = commonViewModel
            commonViewModel.userLoadStatus.subscribe(scope: commonViewModel.viewModelScope) { userLoadStatus in
                self.userLoadStatus = userLoadStatus
            }
        }
        
        func stopObserving() {
            commonViewModel?.dispose()
        }
    }
}

File: androidApp/src/main/java/com/example/stories/android/ui/userData/UserDataViewModel.kt
package com.example.stories.android.ui.userData

import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.example.stories.viewModel.UserDataCommonViewModel

class UserDataViewModel : ViewModel() {

    private val commonViewModel = UserDataCommonViewModel(viewModelScope)
    val userLoadStatus get() = commonViewModel.userLoadStatus
}
